[3006.x] Added support and tests for Sys V service files for RedHat family

dmurphy18 opened the pull request:
### What does this PR do?
Adds service files for salt-minion, salt-master, salt-syndic and salt-api to /etc/init.d

### What issues does this PR fix or reference?
Fixes https://github.com/saltstack/salt/issues/67888

### Previous Behavior
Sys V Service files were missing since 3006.0 release

### New Behavior
Sys V Service files are now present after installation

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**
<!-- Please review the test documentation for details on how to implement tests
into Salt's test suite:
https://docs.saltproject.io/en/master/topics/tutorials/writing_tests.html -->
- [ ] Docs
- [X] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [X] Tests written/updated

### Commits signed with GPG?
Yes

<!-- Please review Salt's Contributing Guide for best practices and guidance in
choosing the right branch:
https://docs.saltproject.io/en/master/topics/development/contributing.html -->

<!-- Additional guidance for pull requests can be found here:
https://docs.saltproject.io/en/master/topics/development/pull_requests.html -->

<!-- See GitHub's page on GPG signing for more information about signing commits
with GPG:
https://help.github.com/articles/signing-commits-using-gpg/ -->

twangboy commented:
Maybe this one needs to go in first...
https://github.com/saltstack/salt/pull/67978

dwoz committed:
Added support and tests for Sys V service files

Rebased onto current 3006.x; manually resolved conflict in
test_salt_upgrade.py (3006.x added _get_installed_salt_packages and
updated test_salt_upgrade signature after the PR's merge-base).
